packages/@aws-cdk/aws-ec2/lib/bastion-host.ts

```diff
@@ -0,0 +1,76 @@
+import { PolicyStatement } from "@aws-cdk/aws-iam";
+import { CfnOutput, Construct } from "@aws-cdk/core";
+import { AmazonLinuxGeneration, AmazonLinuxImage, InstanceClass, InstanceSize, InstanceType } from ".";
+import { Instance } from "./instance";
+import { Port } from "./port";
+import { IVpc, SubnetType } from "./vpc";
+
+/**
+ * Properties of the bastion host
+ */
+export interface BastionHostProps {
+
+    /**
+     * In which AZ to place the instance within the VPC
+     *
+     * @default - Random zone.
+     */
+    readonly availabilityZone?: string;
+
+    /**
+     * VPC to launch the instance in.
+     */
+    readonly vpc: IVpc;
+
+    /**
+     * The name of the instance
+     *
+     * @default - CDK generated name
+     */
+    readonly instanceName?: string;
+
+    /**
+     * Use public subnet instead of private one
+     *
+     * @default - false
```

Comment:
I map this to a standard `SubnetSelection` so yes I could change it. I wanted to make the interface easier for users instead of handling subnet selections. Just tell me if you want it public or not. But I can just change it.